server/src/test/java/org/elasticsearch/search/aggregations/metrics/TopHitsIT.java

```diff
@@ -904,17 +904,14 @@ public void testNestedFetchFeatures() {
         assertThat(highlightField.getFragments().length, equalTo(1));
         assertThat(highlightField.getFragments()[0].string(), equalTo("some <em>comment</em>"));
 
-        // Can't explain nested hit with the main query, since both are in a different scopes, also the nested doc may not
-        // even have matched with the main query.
-        // If top_hits would have a query option then we can explain that query
         Explanation explanation = searchHit.getExplanation();
-        assertFalse(explanation.isMatch());
+        assertTrue(explanation.isMatch());
```

Comment:
A nice side effect of this change is that, because we're finding relevant child queries when we build the SubSearchContext, explanations will now work.